Lamina/include/GL/camera.hpp

```diff
@@ -0,0 +1,145 @@
+#ifndef LM_CAMERA
+#define LM_CAMERA
+
+#include "../Lamina/include/Math/vector.hpp"
+#include "../Lamina/include/GL/window.hpp"
+#include "../Lamina/include/Math/constants.hpp"
+#include "../Lamina/include/user_interface/event.hpp"
+
+namespace lm
+{
+	static lm::mat4 LookAtMatrix(vec3f position, vec3f target, vec3f up)
+	{
+		lm::vec3f direction = lm::normalize(position - target);
+		lm::vec3f right = lm::normalize(lm::CrossProduct(direction, up));
+		lm::mat4 first({
+			{right.x(),		right.y(),		right.z(),		0},
+			{up.x(),		up.y(),			up.z(),			0},
+			{direction.x(),	direction.y(),	direction.z(),	0},
+			{0,				0,				0,				1}
+			});
+		lm::mat4 second({
+			{1, 0, 0, -position.x()},
+			{0, 1, 0, -position.y()},
+			{0, 0, 1, -position.z()},
+			{0, 0, 0, 1}
+			});
+		return first * second;
+	}
+
+	class PivotCamera
+	{
+	public:
+		PivotCamera() : target(lm::vec3f({ 0, 0, 0 })), position(lm::vec3f({ 0, 0, 0 })), radius(5), pitch(0), yaw(0), window(nullptr), pitchClamp(89.f), 
+			radiusClamp(lm::vec2f({5, 50})), firstMouseMovement(true), sensitivity(0.5f), scrollSensitivity(1.0f) {};
+		PivotCamera(lm::vec3f initialPosition, lm::vec3f initialTarget, float initialRadius, lm::Window& _window) :
+			target(initialTarget), position(initialPosition), radius(initialRadius), window(&_window), pitch(0), yaw(0), pitchClamp(89.f), radiusClamp(lm::vec2f({ 5.f, 50.f })),
+		firstMouseMovement(true), sensitivity(0.5f), scrollSensitivity(1.0f)
+		{
+			glfwSetWindowUserPointer(this->window->window, this);
+			glfwSetScrollCallback(this->window->window, lm::PivotCamera::ScrollCallback);
+			glfwSetCursorPosCallback(this->window->window, lm::PivotCamera::CursorPosCallback);
+			UpdateCamera();
+		};
+		PivotCamera(lm::vec3f initialPosition, lm::vec3f initialTarget, float initialRadius, float _pitchClamp, lm::vec2f _radiusClamp, lm::Window& _window) : 
+			target(initialTarget), position(initialPosition), radius(initialRadius), window(&_window), pitch(0), yaw(0), firstMouseMovement(true), sensitivity(0.5f), scrollSensitivity(1.0f)
+		{
+			glfwSetWindowUserPointer(this->window->window, this);
+			glfwSetScrollCallback(this->window->window, lm::PivotCamera::ScrollCallback);
+			glfwSetCursorPosCallback(this->window->window, lm::PivotCamera::CursorPosCallback);
+			if (_radiusClamp.x() > 0.f && _radiusClamp.x() < _radiusClamp.y()) radiusClamp = _radiusClamp;
```

Comment:
It's best to avoid if statements with no brackets since it's very error-prone. You can still keep the if-statement on one line but I'd still keep the bracket.